static/src/javascripts/projects/commercial/modules/comment-adverts.ts

```diff
@@ -47,24 +62,29 @@ const insertCommentAd = (
 };
 
 const containsDMPU = (ad: Advert): boolean =>
-	ad.sizes.desktop.some(
+	!!ad.sizes.desktop?.some(
 		(el) =>
 			(el[0] === 300 && el[1] === 600) ||
 			(el[0] === 160 && el[1] === 600),
 	);
 
 const maybeUpgradeSlot = (ad: Advert, adSlot: Element): Advert => {
-	if (!containsDMPU(ad)) {
+	if (!containsDMPU(ad) && ad.sizes.desktop) {
 		const extraSizes: SizeKeys[] = ['halfPage', 'skyscraper'];
 		ad.sizes.desktop.push(
 			// TODO: add getTuple method to commercial-core
 			...extraSizes.map((size) => {
 				const { width, height } = adSizes[size];
-				const tuple: AdSizeTuple = [width, height];
+				const tuple = createAdSize(width, height);
 				return tuple;
 			}),
 		);
-		ad.slot.defineSizeMapping([[[0, 0], ad.sizes.desktop]]);
+		const sizeMapping = ad.sizes.desktop.map((size) =>
+			!size.width && !size.width
```

Comment:
I think `&& !size.width` can be removed here

Reply:
good catch, it should be `!size.width && !size.height` actually!